crates/turborepo-lib/src/config.rs

```diff
@@ -0,0 +1,132 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::Result;
+use config::{Config, Environment};
+// Go's xdg implementation uses FOLDERID_LocalAppData for config home
+// https://github.com/adrg/xdg/blob/master/paths_windows.go#L28
+// Rust xdg implementations uses FOLDERID_RoamingAppData for config home
+// We use cache_dir so we can find the config dir that the Go code uses
+#[cfg(windows)]
+use dirs_next::cache_dir as config_dir;
+#[cfg(not(windows))]
+use dirs_next::config_dir;
+use serde::{Deserialize, Serialize};
+
+pub fn default_user_config_path() -> Option<PathBuf> {
+    config_dir().map(|p| p.join("turborepo").join("config.json"))
```

Comment:
If this `Option` being `None` is always an error, it might be better to return the error from here

```suggestion
pub fn default_user_config_path() -> Result<PathBuf> {
    config_dir().map(|p| p.join("turborepo").join("config.json")).context("default config path not found")
```

same goes for `data_dir` below